### ISSUE TITLE
Is it possible to send metric data to ES instead of RRD

### ISSUE THREAD

**Issue by xmudall** [ganglia/monitor-core#250]
Hi all,

I wonder that can gmetad sends data to elasticsearch other than RRDtool. I'd like to store data on ES and use Kibana as dashboard.

**Comment by umairpannu** [comment]
Yes you can but you will have to write a custom script for this. I have written a program that gets XML dump from gmetad and pushed the metrics to opentsdb.

**Comment by xmudall** [comment]
Wonderful! 
Would you plz give me some hints about the program/script or/and where to start it? Thanks a lots.

**Comment by umairpannu** [comment]
Mine setup worked as fallowed I installed the monitoring demon on every node and in /etc/conf/gmond configured it to send metrics to the local host only then I installed my custom program on all nodes that takes XML dump from the monitoring demon parse it and then send to opentsdb, if you want I can share the scala code with you. And all the aggregation was done on the opentsdb side.

**Comment by xmudall** [comment]
I got it. It's most like a custom gmetad. I'm not familiar with scala, i
think i'll rewrite it in java or python.
So kind you are , thanks a lot~

On Wed, Mar 23, 2016 at 1:04 PM, umairpannu <notifications@github.com>
wrote:

> Mine setup worked as fallowed I installed the monitoring demon on every
> node and in /etc/conf/gmond configured it to send metrics to the local host
> only then I installed my custom program on all nodes that takes XML dump
> from the monitoring demon parse it and then send to opentsdb, if you want I
> can share the scala code with you. And all the aggregation was done on the
> opentsdb side.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ganglia/monitor-core/issues/250#issuecomment-200184073>
>



-- 

    祝
好

胡游乐
清华大学软件学院
xmudall@gmail.com

**Comment by hawson** [comment]
While not ElasticSearch, I've been working on sending metrics to InfluxDB.  See https://github.com/hawson/monitor-core/tree/influxdb for details.

What format does ElasticSearch use?

The idea of scraping the local XML feed is a clever one, but possibly heavyweight as compared to just sending the data directly, without need for the XML serialization (relatively expensive), parsing and translation (possibly also expensive), and re-sending.
